### ISSUE TITLE
bug: pushing an image up to localstack `ecr`; `awslocal ecr batch-get-images` says ImageNotFound?

### ISSUE THREAD

**Issue by iiian** [localstack/localstack#5598]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When trying to push an container image up to `localstack` via docker, and based on the advertised localstack.cloud documentation, I am then faced with multiple clues telling me that the image has in fact not been pushed to `localstack`.

### Expected Behavior

Localstack takes my container image, and I can use it to invoke a lambda function.

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

file: `docker-compose.yml`
```yml
version: "3.8"

services:
  localstack:
    image: localstack/localstack:latest
    environment:
      - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY-}
      - EDGE_PORT=4566
      - LAMBDA_EXECUTOR=docker
      - LAMBDA_REMOTE_DOCKER=0
      - SERVICES=rds,vpc,lambda,dynamodb,cloudformation,s3,sts,iam,apigateway,ecr,ec2
      - HOST_TMP_FOLDER="${TMPDIR:-/tmp}/localstack"
      - DEFAULT_REGION=us-west-1
      - PORT_WEB_UI=8080
    ports:
      - '4510-4599:4510-4599'
      - '8080:8080'
    volumes:
      # - '${TMPDIR:-/tmp/localstack}:/tmp/localstack'
      - '/var/run/docker.sock:/var/run/docker.sock'
    privileged: true
```

cloudformation file: `template.yml`:
```yml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  DocSvcCreateDoc:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        ImageUri: localhost:4510/createdoc
```

Steps:
1. `docker-compose up`
2. In a separate terminal, once the localstack application is stood up:
```bash
$ awslocal ecr create-repository --repository-name createdoc;
$ docker build -t localhost:4510/createdoc .;
$ docker push localhost:4510/createdoc;
$ awslocal cloudformation create-stack --stack-name test --template-body file://./template.yml;
```
None of the above commands fail. (1) The repository claims to be created, (2) I can turn around and inspect that it's been created, (3) `docker push localhost:4510/createdoc` has the tell-tale racing arrows of a regular docker push, (4) `create-stack` returns figures about a stack thats been created and the `localstack` log process shows activity like somethings being done after that command.

However:
1. As I am "uploading" my docker image to localstack, the **localstack** process (not to be confused with the process running docker push) shows no indication is log output that it is even frankly receiving anything.
2. When I try to `awslocal lambda invoke` my "created" function, I am met with an error in both terminals:
```log
localstack_1  |     container_id = DOCKER_CLIENT.create_container(
localstack_1  |   File "/opt/code/localstack/localstack/utils/container_utils/docker_sdk_client.py", line 535, in create_container
localstack_1  |     raise ContainerException()
localstack_1  | localstack.utils.container_utils.container_client.ContainerException
localstack_1  | 2022-03-03T08:36:08.009:INFO:localstack_ext.services.awslambda.lambda_extended: Lambda container output: None
localstack_1  | None
```
3. When I: 
```bash
$ awslocal ecr batch-get-images \
   --repository-name createdoc \
   --image-id \
      imageDigest=<image-digest-for-localhost:4510/createdoc>,\
      imageTag=latest`:
```
I am met with:
```JSON
{
    "images": [],
    "failures": [
        {
            "imageId": {
                "imageDigest": "<image-digest-for-localhost:4510/createdoc>",
                "imageTag": "latest"
            },
            "failureCode": "ImageNotFound",
            "failureReason": "Requested image not found"
        }
    ]
}
```

Which is disheartening. Although, I get conflicting results from `list-images`:

```bash
$ awslocal ecr list-images --repository-name createdoc
```

yields

```JSON
{
    "imageIds": [
        {
            "imageDigest": "sha256:<image-digest-for-localhost:4510/createdoc>",
            "imageTag": "latest"
        }
    ]
}
```

So I'm not sure. It could be operator error. Let me know what you guys think.

### Environment

```markdown
- OS: Ubuntu 20.04 + WSL@2.0 (Windows 10)
- LocalStack: latest
```

### Anything else?

You guys are awesome.

**Comment by iiian** [comment]
Now on the `latest` version, it just seems to balk at anything over 750MB.

**Comment by iiian** [comment]
Which seems to be related to [this line in lambda_api.py](https://github.com/localstack/localstack/blob/a7602821ac8666c9ab3a253457eb99570908ac8e/localstack/services/awslambda/lambda_api.py#L95)?

**Comment by iiian** [comment]
`FUNCTION_MAX_SIZE = 69905067`
and then later
```python
if size > FUNCTION_MAX_SIZE:
  raise RequestEntityTooLargeException()
```

**Comment by whummer** [comment]
Thanks for reporting @iiian . We have now pushed a few fixes that should make ECR more robust to larger image sizes. Can you please pull the `latest` LocalStack Docker image and give it another try? Please let us know if the problem persists.. Thanks for your help!

**Comment by iiian** [comment]
No dice. Seems to be the same issue as before. My upload attempt results in:

```bash
$ docker push localhost:4510/createdoc
...
error parsing HTTP 413 response body: invalid character '<' looking
for beginning of value: "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 
3.2 Final//EN\">\n<title>413 Request Entity Too 
Large</title>\n<h1>Request Entity Too Large</h1>\n<p>The data value 
transmitted exceeds the capacity limit.</p>\n"
```

**Comment by whummer** [comment]
Thanks for the update @iiian . Just to confirm the version - can you please share the logs that gets printed during startup? Thanks

Just gave it a try with the latest image:
```
LocalStack version: 0.14.0
LocalStack Docker container id: 7bc85e8d7c66
LocalStack build date: 2022-03-07
LocalStack build git hash: 951073bf
```
... creating an ECR repo and pushing a ~2GB large image:
```
$ awslocal ecr create-repository --repository-name createdoc
...
$ docker push localhost:4510/createdoc
... 
latest: digest: sha256:fb52... 
```

**Comment by iiian** [comment]
BOOM BABY! It's fixed. Thanks so much.

**Comment by iiian** [comment]
whoops, sorry didn't mean to close it, my issue is resolved but don't let me disrupt your regular bug-mgmt process.

**Comment by ptrhck** [comment]
I am trying to push an image with >8GB in size and get the following error:
`error parsing HTTP 408 response body: invalid character '<' looking for beginning of value: "<!doctype html>\n<html lang=en>\n<title>408 Request Timeout</title>\n<h1>Request Timeout</h1>\n<p>The server closed the network connection because the browser didn&#x27;t finish the request within the specified time.</p>\n"`
Is this related and how can this be fixed? I am on `0.14.3`.
